### DISCUSSION TITLE
DatabaseStatus not found

### DISCUSSION THREAD

**Discussion by FeltenVanballenberghe** [FraunhoferIOSB/FROST-Server#1441] [Q&A] [answered]
I can't seem to get the database connection to work
Debug or guidance would be kind.

STACK//
- debian 11
- tomcat 10.1.5.
- FrostServer [FROST-Server.MQTTP] standard .war file
- Postgres with postgres gis 14 (remote and local)
- postgis-jdbc-2.4.0.jar
- postgresql-42.2.16.jar

//Resource in 
context.xml (done in webapps/..Meta-INF/context.xml or /tomcat/[engine]/localhost/frostserver.xml)

 <Resource
        name="jdbc/sensorThings" auth="Container"
        type="javax.sql.DataSource" driverClassName="org.postgresql.Driver"
        url="jdbc:postgresql://localhost:5432/sens"
        username="correct" password="correct"
        maxTotal="20" maxIdle="10" maxWaitMillis="-1"
        defaultAutoCommit="false"
    />

//Postgres
The remote and the local are reachable and have a working database with gis-extension installed

//PROBLEM
tomcat/FrostServer is working
**tomcat/FrostServer/DatabaseStatus (GET localhost:8080/Frostserver/databaseStatus) gives 404**

What have i been doing wrong?
How can I debug this for feature errors?
Everythings runs in proxmox and its reachable true putty/terminal/webbrowser

Thx in advance!

*edit i'm a dotNet baby - not a java child - unfortunaly*

**Comment by FeltenVanballenberghe** [comment]
![image](https://user-images.githubusercontent.com/83814030/219871436-fdc114fd-b0e0-4fe6-8181-004e3c2dc2dc.png)

**Comment by hylkevds** [comment, marked as answer]
FROST-Server currently still uses version 4.0 of the Servlet specification, and is thus not compatible with Tomcat 10.
In the Tomcat log file you should see a lot of FROST initialisation output if FROST is working. If you see not output, FROST is not being initialised.

**Reply by FeltenVanballenberghe** [reply]
Thank you sir! Got it running

**Comment by FeltenVanballenberghe** [comment]
For future reference:
- Stack used: Tomcat/9.0.71 -- JVM 17+35 -- postgres and postgis 14.x
- Set a setenv.sh file in the $catalina\bin to adjust  JAVA_HOME (sudo update-alternatives  --config is a great tool)
- context.xml uncomment  parameter "serviceRootUrl"
- tail -f $catalina\log\localhost.<dateTime>.log ==> for more container info
